src/_pytest/logging.py

```diff
@@ -613,7 +609,7 @@ def _runtest_for(self, item, when):
                 yield
 
     @contextmanager
-    def _runtest_for_main(self, item, when):
+    def _runtest_for_main(self, item, when: str) -> Generator[None, None, None]:
```

Comment:
I *would* type this as `Item`, and add gitignores to the custom attribute get/sets. For the caller `Item` is the correct type, and the ignores are also correct insofar as static typing goes.